src/Stratis.Bitcoin.Features.SmartContracts/ReflectionExecutor/Controllers/SmartContractsController.cs

```diff
@@ -329,9 +331,18 @@ public IActionResult BuildAndSendCreateSmartContractTransaction([FromBody] Build
                 return this.Json(response);
 
             Transaction transaction = this.network.CreateTransaction(response.Hex);
-            this.walletManager.ProcessTransaction(transaction, null, null, false);
+
             this.broadcasterManager.BroadcastTransactionAsync(transaction).GetAwaiter().GetResult();
 
+            // Check if transaction was actually added to a mempool.
+            TransactionBroadcastEntry transactionBroadCastEntry = this.broadcasterManager.GetTransaction(transaction.GetHash());
```

Comment:
It looks like this code and the changes below are the same? Perhaps put it in a shared method?

Reply:
I thought about it already, but there is not much benefit. Controllers don't share common base class and there wait it constructs a result would not improve code that much.